3di quickstart updates

awie3di opened the pull request:
<!---
Notes:
- Your PR should conform to our [Contributing Guidelines](https://github.com/auth0/docs/blob/master/CONTRIBUTING.md)
- If applicable, add details to the [update feed](https://github.com/auth0/docs/tree/master/updates)
- Make sure your PR gets deployed to a Heroku [Review App](https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#review-apps) without errors.
- Please include a short description
- If your PR is a work in progress title it [DO NOT MERGE]
--->
These are the changes we propose for the following quickstarts: 
Angular 2+ 
React 
AngularJS
Node.js
JavaScript

awie3di committed:
Angular 2+: Login and User Profile

Initial changes to the Login article in the Anguler 2+ QSG

awie3di committed:
Angular 2+ Initial changes

awie3di committed:
Correcting the Angular2+ Login article

awie3di committed:
Correcting the Angular 2+ User Profile article

awie3di committed:
Correcting the Angular 2+ Calling an API article

awie3di committed:
Correcting the Angular 2+ Authorization article

awie3di committed:
Correcting the Angular 2+ Token Renewal article

awie3di committed:
Corrected the User Profile article

awie3di committed:
Rewriting the React Login article

awie3di committed:
Aligning the Angular Login article with the React Login article

awie3di committed:
Rewriting tha React User Profile article

awie3di committed:
Aligning the Angular User Profile article with the React User Profile article

awie3di committed:
Rewriting the React Calling an API Article

awie3di committed:
Aligning the Angular Calling an API article with the React Calling an API article

awie3di committed:
Rewriting the React Authorization article

awie3di committed:
Aligning the Angular 2+ Authorization article with the React Authorization article

awie3di committed:
Correcting the React Login article

awie3di committed:
Correcting the React User Profile article

awie3di committed:
Correcting the React Creating an API article

awie3di committed:
Correcting the React Authorization article

awie3di committed:
Correcting the React Calling an API article

awie3di committed:
Correcting the Angular 2+ quickstart after peer review

awie3di committed:
Correcting the React Quickstart after peer review

awie3di committed:
Creating a snippet for 'User Profile: Add a Profile Component)

awie3di committed:
Creating a snippet for Calling an API: Set the Audience and Scope

awie3di committed:
Creating a snippet for Calling an API: try to log in + use rules

awie3di committed:
Creating a snippet for Authorization: Handle scopes in the service

awie3di committed:
Creating a snippet for Authorization: Conditionally display UI elements

awie3di committed:
Creating a snippet for Authorization: Protecting Client-Side Routes

awie3di committed:
Correcting a panel

awie3di commented on articles/quickstart/spa/_includes/_calling_api_create_scope.md:
```diff
@@ -1,7 +1,11 @@
 ## Add a Scope
 
-When an `access_token` is received after a user authenticates in your app, it is simply an indication that they have proven their identity. By default, however, the `access_token` does not contain any **authorization** information. To properly limit access to your resources based on authorization, you must use [scopes](https://auth0.com/docs/api-auth/tutorials/adoption/api-tokens).
+By default, the access token does not contain any authorization information. To limit access to your resources based on authorization, you must use scopes. Read more about scopes in the  [api tokens documentation](https://auth0.com/docs/api-auth/tutorials/adoption/api-tokens).
```
I left the link as is, but shouldn't it be just (/api-auth/tutorials/adoption/api-tokens)?

mpaktiti replied:
@awie3di Yes all links should be relative, excluding the `https://auth0.com/docs` part. So this link should be `Read more about scopes in the  [api tokens documentation](/api-auth/tutorials/adoption/api-tokens).`

mpaktiti replied:
I would update this link to point to this doc https://auth0.com/docs/scopes

awie3di commented on articles/quickstart/spa/_includes/_calling_api_create_scope.md:
```diff
@@ -1,7 +1,11 @@
+By default, the access token does not contain any authorization information. To limit access to your resources based on authorization, you must use scopes. Read more about scopes in the  [api tokens documentation](https://auth0.com/docs/api-auth/tutorials/adoption/api-tokens).
 
-In the settings area for your API within the Auth0 dashboard, navigate to the **Scopes** tab. Add any scopes necessary to limit access to your API resources. While the naming of the scopes is at your discretion, a common pattern is to use `<action>:<resource>`. The example here will use a `read:messages` scope.
+In the Auth0 dashboard, in the **APIs** section, click **Scopes**. Add any scopes you need to limit access to your API resources.
```
We could add a link to the APIs section of the dashboard. 

Not sure about the exact UI label: is it APIs section or API section?

mpaktiti replied:
We use `APIs` since that's what the left-hand menu reads. You can add a link using `${manage_url}/#/apis`

(the variable `${manage_url} translates to `https://manage.auth0.com`)

awie3di commented on articles/quickstart/spa/_includes/_calling_api_preamble.md:
```diff
@@ -1,3 +1,7 @@
-It's likely that your single page app will need to consume resources from a data API. Since you are adding authentication to your app, presumably you need to limit access to these resources such that they are only accessible by an authenticated user who is authorized to retrieve them. Auth0 provides [API Authorization](https://auth0.com/docs/api-auth) to accomplish this.
+Some single-page apps use resources from data APIs. You may want to restrict access to those resources, so that only authenticated users with sufficient privileges can access them. Auth0 lets you manage access to these resources using [API Authorization](https://auth0.com/docs/api-auth).
```
Should the link be in this form? According to the 'contributing' guide, this link should be in the (/api-auth) form.

mpaktiti replied:
You are right, it should be `/api-auth`

awie3di commented on articles/quickstart/spa/_includes/_calling_api_use_rules.md:
```diff
@@ -0,0 +1,7 @@
+:::
+
+::: note
+By default, any user can ask for any scope you defined. You can implement access policies to limit this behavior using [Rules](https://auth0.com/docs/rules).
```
I didn't change any links that were originally in the content, but the contributing guide mentions that they should be relative.

mpaktiti replied:
Please change them. And thanks for pointing that out!

awie3di commented on articles/quickstart/spa/angular2/05-token-renewal.md:
```diff
@@ -69,15 +76,17 @@ console.log('Listening on http://localhost:3001');
 
 ::: note
-Be sure to add `http://localhost:3001/silent` to the **Callback URLs** section in your application's client settings.
+Add `http://localhost:3001/silent` to the **Callback URLs** section in your application's client settings.
```
Maybe we could add a link to the mentioned section?

mpaktiti replied:
Actually this is not a section. It is a field in the Client Settings:

go to Dashboard > Clients > select a client > Settings > the field is named `Allowed Callback URLs`

![image](https://user-images.githubusercontent.com/11715799/29560099-a8e13140-8739-11e7-895a-5ac0c8efa3ac.png)

mpaktiti replied:
https://github.com/auth0/docs/pull/4848/files#diff-e9b2c9e7ab44127591bed2d47f927119R5

awie3di commented on articles/quickstart/spa/angular2/05-token-renewal.md:
```diff
@@ -69,15 +76,17 @@ console.log('Listening on http://localhost:3001');
 
 ::: note
-Be sure to add `http://localhost:3001/silent` to the **Callback URLs** section in your application's client settings.
+Add `http://localhost:3001/silent` to the **Callback URLs** section in your application's client settings.
```
Maybe we could add a link to the mentioned section?

awie3di commented on articles/quickstart/spa/angular2/_includes/_centralized_login.md:
```diff
@@ -1,12 +1,24 @@
+You can also embed the Lock widget directly in your application. If you use this method, some features, such as single sign-on, will not be accessible. 
+To learn how to embed the Lock widget directly in your application, follow the [Embedded Login sample](https://github.com/auth0-samples/auth0-angular-samples/tree/embedded-login/01-Embedded-Login).
+:::
+
```
This section was originally on the bottom of the page. I moved it to the top because someone who decides to use the Lock widget should not have to read through the whole section before they get to this information.

awie3di commented on articles/quickstart/spa/angular2/_includes/_centralized_login.md:
```diff
@@ -37,15 +49,21 @@ export class AuthService {
 
-Add some additional methods to the `AuthService` to fully handle authentication in the app.
+Add more methods to the `AuthService` service to handle authentication in the app.
+
```
The code sample below is quite long. Could it be separated into four sections (a section for a method)? 
Alternatively, all the code samples could have comments describing what each section does.

awie3di commented on articles/quickstart/spa/_includes/_auth_service_method_description_auth0js.md:
```diff
@@ -1,7 +1,30 @@
+* Audience and scope, which specify that authentication must be [OIDC conformant](https://auth0.com/docs/api-auth/tutorials/adoption)
+* The URL where your users are redirected to after authentication.
+::: note
+In this tutorial, the route is `/callback`, which is implemented in the Add a Callback Component step. 
```
Do you use bookmarks in your content? We could add a link to the 'Add a Callback Component' step (further down in this article) here.

awie3di commented on articles/quickstart/spa/react/_includes/_centralized_login.md:
```diff
@@ -130,22 +140,28 @@ Provide a component with controls for the user to log in and log out.
 
 <%= include('../../_includes/_hosted_login_customization' }) %>
 
 ## Add a Callback Component
```
In the Angular 2+ guide, the order of the 'Process authentication result' and 'Add a callback component' steps is reversed: 

Angular 2+:
Process result
Add a component

React: 
Add a component
Process result

Was that intentional? Or could we unify the order of those steps?

aaguiarz replied:
The 'add a callback component' should be before 'Process Result' in both

awie3di committed:
Merge branch 'master' into 3di-quickstart-updates

awie3di commented on articles/quickstart/spa/angular2/02-user-profile.md:
```diff
@@ -17,7 +17,7 @@ budicon: 292
 ## Request the Profile Scope
 
-The user's `access_token` requires a `scope` of `openid profile` to successfully retrieve their information. In the `WebAuth` instance, specify that you would like to ask for these scopes.
+To retrieve user information, request a scope of `openid profile` in the `WebAuth` instance. 
```
@aaguiarz is WebAuth the same thing as auth0.WebAuth? If yes, I would suggest unifying the way it is mentioned. Also, is the instance of the webauth object always the same one or do you create multiple instances?

aaguiarz replied:
Let's use auth0.WebAuth. In general you create a single one, if you see an example where there are more than one, let me know.

awie3di commented on articles/quickstart/spa/_includes/_calling_api_create_scope.md:
```diff
@@ -1,7 +1,11 @@
+You can give any names to your scopes. `<action>:<resource>` is a common pattern. The example below uses the name `read:messages` for a scope.
+:::
 
 ![create scope](/media/articles/api-auth/create-scope.png)
```
@aaguiarz the screenshot shows the same note that is in the content. I rephrased the note. Would it be possible to change the content of the note that appears in the interface as well? The revised note says:
By default, any user can ask for any scope you defined. You can implement access policies to limit this behavior using Rules(link).

awie3di commented on articles/quickstart/spa/angular2/04-authorization.md:
```diff
@@ -115,11 +109,13 @@ export const ROUTES: Routes = [
-To prevent access to client-side routes based on a particular `scope`, create another service called `ScopeGuard`. This service should use `ActivatedRouteSnapshot` to check for a set of `expectedScopes` passed in the `data` key of the route configuration.
+To prevent access to client-side routes based on a scope, create a service called `ScopeGuard`. This service uses the  `ActivatedRouteSnapshot` method to check for a set of `expectedScopes` passed in the `data` key of the route configuration.
+
+If the user does not have the `write:messages` scope, they are redirected to the main route.
```
@aaguiarz  Is the main route the same as /home route?

aaguiarz replied:
Not necessarily, I'd leave the content as it is.

awie3di commented on articles/quickstart/spa/angular2/04-authorization.md:
````diff
@@ -115,11 +109,13 @@ export const ROUTES: Routes = [
 ```
 
-The guard implements the `CanActivate` interface which requires a method called `canActivate` in the service. This method returns `true` if the user is authenticated and `false` if not. It also navigates the user to the home route if they aren't authenticated.
+The guard implements the `CanActivate` interface which requires a method called `canActivate` in the service. This method returns `true` if the user is authenticated and `false` if they are not. It also navigates the user to the home route if they are not authenticated.
````
@aaguiarz  Just a thought, could the interface have a different name? The difference between the interface name and the method name is just a capital 'C'. Wouldn't it potentially cause problems to spot errors in code?

aaguiarz replied:
The interface is from the Angular core, so we can't change it (https://angular.io/api/router/CanActivate)

awie3di commented on articles/quickstart/spa/_includes/_authz_handle_scopes.md:
```diff
@@ -0,0 +1,3 @@
+## Handle Scopes in the ${service} service
+
+Adjust your ${service} service to use a local member with any scopes you want to request when users log in. Use this member in your instance of the `auth0.WebAuth` object.
```
@aaguiarz, could you help me understand what this sentence means? It's not clear so I'm not sure how can I correct it. Is it: 
Adjust the service with scopes, so it can use a local member
or
Adjust the service, so it uses a local member with scopes

mpaktiti replied:
Using this `${service}` is wrong (and it might be the reason why the Heroku review app fails). You can only use this for registered repo variables https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#document-variables 

If you want to display this format, you need to escape it, see here how https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#escaping-strings

awie3di replied:
As discussed on slack, this is a placeholder that is replaced when the snippet is inserted into an article (the variable is defined in the insert command).

aaguiarz replied:
The meaning is "Adjust the service, so it uses a local member with scopes"

awie3di committed:
Rewriting the Login article in the Nodejs quickstart.

awie3di committed:
Creating snippets about the Auth0 hosted login page

awie3di committed:
Updating the content in the Token Renewal topic in the Angular 2+ and React articles

awie3di committed:
Rewriting the User Profile article in the Nodejs guide

awie3di committed:
Adding a link to Client Settings (Login; Nodejs)

mpaktiti commented on articles/_includes/_login_auth0_hosted_login_page.md:
```diff
@@ -0,0 +1 @@
+The [Auth0 hosted login page](https://auth0.com/docs/hosted-pages/login) is the easiest way to set up authentication in your application. 
```
Make URL relative `[Auth0 hosted login page](/hosted-pages/login)`

mpaktiti commented on articles/quickstart/spa/_includes/_auth_service_method_description_auth0js.md:
```diff
@@ -1,7 +1,30 @@
+When you set up the `AuthService` service, you create an instance of the `auth0.WebAuth` object. In that instance, you can define the following:
+* Configuration for your client and domain
+* Response type, to show that you need a user's access token and an ID token after authentication
+* Audience and scope, which specify that authentication must be [OIDC-conformant](https://auth0.com/docs/api-auth/tutorials/adoption)
```
Make the URL relative and update it to point to this doc https://auth0.com/docs/api-auth/intro

I would also change the wording to something `Audience and scope, which enable you to use the [latest and most secure Auth0 authentication pipeline](/api-auth/intro)`. Actually only `audience` affects that, not scope. You can set scope and still use the legacy pipeline.

mpaktiti replied:
I don't know if our quickstarts have a panel or something to inform the users what they should do to use the latest pipeline (enable the OIDC conformant flag or use audience), but maybe they should.

cc @aaguiarz

mpaktiti commented on articles/quickstart/spa/_includes/_auth_service_method_description_auth0js.md:
```diff
@@ -1,7 +1,30 @@
+* Audience and scope, which specify that authentication must be [OIDC-conformant](https://auth0.com/docs/api-auth/tutorials/adoption)
+* The URL where your users are redirected to after authentication.
+::: note
+In this tutorial, the route is `/callback`, which is implemented in the Add a Callback Component step. 
```
This could probably use a link to the step (Add a Callback Component)

awie3di replied:
Yes, i left it here with the intention to add a link. I was just wondering if you use bookmarks in the content. Will correct.

aaguiarz replied:
Markdown always adds bookmarks for headers, so you can link to them (e.g. https://auth0.com/docs/quickstart/spa/angular2#add-a-callback-component)

mpaktiti commented on articles/quickstart/spa/_includes/_auth_service_method_description_auth0js.md:
```diff
@@ -1,7 +1,30 @@
+* `id_token`
+* `expires_in`
+
+To log the user out, remove these values from the storage. 
```
This is not enough though right? They have to do this to terminate the application session, but they also need to terminate the Auth0 session as well. I would clarify this a bit.

mpaktiti commented on articles/quickstart/spa/_includes/_auth_service_method_description_auth0js.md:
```diff
@@ -1,7 +1,30 @@
+To log the user out, remove these values from the storage. 
+
+::: note
+Our examples use local storage to save the user's authentication information. You can also use session storage or cookies.
```
I would point here for more info https://auth0.com/docs/security/store-tokens

mpaktiti commented on articles/quickstart/spa/_includes/_authz_client_routes_disclaimer.md:
```diff
@@ -1,9 +1,9 @@
+For the access control on the client-side, the `scope` values that you get in local storage are only a clue that the user has those scopes. The user could manually adjust the scopes in local storage to access routes they shouldn't have access to. 
 
-So how are you supposed to create a secure application if client-side routes are so easily hackable? The answer is to remember that browsers are public clients and that they must be treated as such. Any and all sensitive data which powers your application needs to be kept on your server instead of being included directly in your client-side SPA because your server is the only place that can act a ...
+::: panel-warning Do not store your sensitive data in your SPA
```
The idea for panels is that we use `warning` if the text is up to 4 lines. For longer text we use `panel-warning`. Same with `note` vs `panel`

https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#ui-components

mpaktiti commented on articles/quickstart/spa/_includes/_authz_preamble.md:
```diff
@@ -1,16 +1,16 @@
 ::: note
-Auth0 makes no assumptions about how the `scope`s for your API map to various access levels, nor is there any restriction on what you call your `scope`s and access levels. These details are at your discretion.
-:::
+You can give your scopes any names and map them to any access levels you want. 
```
No you cannot give any names you want. You cannot use standard claims names, and you need to namespace the custom ones 

https://auth0.com/docs/scopes

https://auth0.com/docs/api-auth/tutorials/adoption/scope-custom-claims

aaguiarz replied:
Maybe something like:

You can give your scopes any names and map them to any access levels you want, making sure you follow the rules described (here)[https://auth0.com/docs/scopes]

@mpaktiti?

awie3di replied:
How about omitting the rather problematic 'any name you want' and saying:
"Read about naming scopes and mapping them to access levels in the [Scopes documentation](https://auth0.com/docs/scopes). "

To give more detail, I could add something like: 
"To learn more about standard and custom scope claims (is that the right term?), follow the [tutorial](https://auth0.com/docs/api-auth/tutorials/adoption/scope-custom-claims)."

mpaktiti commented on articles/quickstart/spa/_includes/_authz_protect_client_side_routes.md:
```diff
@@ -0,0 +1,3 @@
+## Protect Client-Side Routes
+
+You may want to give access to some routes in your application only to authenticated users. You can check if the user is authenticated with the ${check}.
```
This will cause a build error, you need to escape `${check}` https://github.com/auth0/docs/blob/master/CONTRIBUTING.md#escaping-strings

awie3di replied:
My intention here was to use a variable when inserting a snippet (depending on the guide, the method for checking is called differently). Is that the wrong format for entering variables?

awie3di replied:
As discussed on slack, this is a placeholder that is replaced when the snippet is inserted into an article (the variable is defined in the insert command).

mpaktiti commented on articles/quickstart/spa/_includes/_authz_set_session.md:
```diff
@@ -1 +1,7 @@
-In the `setSession` method, save the `scope`s granted for the user into local storage. The first place to check for these granted `scope` values is the `scope` key from the `authResult`. If something exists there it's because the `scope`s which were granted for the user differ from those that were requested. If there is nothing on `authResult.scope`, it means that the granted `scope`s match those ...
\ No newline at end of file
+Check for the scopes granted for the user. 
```
granted `to` the user

mpaktiti commented on articles/quickstart/spa/_includes/_callback_component.md:
```diff
@@ -1,3 +1,6 @@
-* It prevents the need to whitelist multiple (and sometimes unknown) callback URLs
-* It serves as a place to show a loading indicator while your application sets the user's client-side session
+::: panel Select a callback URL
+You can select any URL in your application for your users to return to. We recommend creating a dedicated callback route. 
```
I'd leave it as `we recommend`

mpaktiti commented on articles/quickstart/spa/_includes/_calling_api_create_scope.md:
```diff
@@ -1,7 +1,11 @@
+In the Auth0 dashboard, in the **APIs** section, click **Scopes**. Add any scopes you need to limit access to your API resources.
+
+::: note
+You can give any names to your scopes. `<action>:<resource>` is a common pattern. The example below uses the name `read:messages` for a scope.
```
I would avoid saying `you can give any names to your scopes`. See this https://auth0.com/docs/scopes/current#custom-claims

awie3di replied:
I will align this note with the note from articles/quickstart/spa/_includes/_authz_preamble.md

mpaktiti commented on articles/quickstart/spa/_includes/_calling_api_preamble.md:
```diff
@@ -1,3 +1,7 @@
 
-This tutorial will guide you through how to make calls from your application to your data API to access protected resources. It does not, however, cover how to add protection to your API itself. For instructions on protecting your API, please follow the [Backend/API quickstart documentation](quickstart/backend) specific to your situation.
\ No newline at end of file
+This tutorial shows you how to access protected resources in your data API. 
```
I haven't read the complete doc from start to finish, just fragments, but I don't get the `data API`. Why not `This tutorial shows you how to access protected resources in your API.`?

mpaktiti commented on articles/quickstart/spa/_includes/_calling_api_protect_resources.md:
```diff
@@ -1,7 +1,12 @@
 
-To restrict access to the resources served by your API, a check needs to be made to determine whether the incoming request contains valid authorization information. When you implement Auth0 in your client-side application and data API, that authorization information is the `access_token` issued for your user. This token must be checked against the [JSON Web Key Set (JWKS)](https://auth0.com/blog/ ...
+To restrict access to the resources served by your API, check the incoming requests for valid authorization information. 
+The authorization information is in the `access_token` created for the user. To see if the token is valid, check it against the [JSON Web Key Set (JWKS)](https://auth0.com/blog/navigating-rs256-and-jwks/) for your Auth0 account.
```
I would link to this doc instead https://auth0.com/docs/jwks

mpaktiti replied:
we also have a separate doc for validating access tokens, I'd link to that https://auth0.com/docs/api-auth/tutorials/verify-access-token

mpaktiti commented on articles/quickstart/spa/_includes/_calling_api_use_rules.md:
```diff
@@ -0,0 +1,7 @@
+::: panel Checkpoint
+Try to log in to your application again. Look at how the `access_token` differs from before. It is no longer an opaque token. Instead, it is now a JSON Web Token with a payload that contains your API identifier as audience and the scopes you created.
```
I'd add a link either to https://auth0.com/docs/tokens/access-token or to https://auth0.com/docs/jwt

mpaktiti replied:
Also, this should be a `note`, not a `panel` (<= 4 lines)

mpaktiti commented on articles/quickstart/spa/_includes/_getting_started.md:
```diff
@@ -1,13 +1,16 @@
-This integration guide will walk you through setting up and managing authentication and authorization in your ${library} apps using Auth0.
+This guide walks you through setting up authentication and authorization in your ${library} apps with Auth0.
```
`${library}` will cause a build error

awie3di replied:
As discussed on slack, this is a placeholder that is replaced when the snippet is inserted into an article (the variable is defined in the insert command).

mpaktiti commented on articles/quickstart/spa/_includes/_getting_started.md:
````diff
@@ -1,13 +1,16 @@
+If you are following along with the sample project you downloaded from the top of this page,  **Callback URL** should be set to
 
 ```bash
 ${callback}
````
The `${callback}` will cause a build error, you have to escape it. Also, are you sure you don't want to use `${account.namespace}`?

mpaktiti replied:
Since this is a note remove the whitespace and set it like this:

If you are following along with the sample project you downloaded from the top of this page,  **Callback URL** should be set to `${callback}`.

(note my previous comment also, this is not a valid variable name)

awie3di replied:
As discussed on slack, this is a placeholder that is replaced when the snippet is inserted into an article (the variable is defined in the insert command).

mpaktiti commented on articles/quickstart/spa/_includes/_hosted_login_customization.md:
```diff
@@ -1,3 +1,3 @@
 ::: note
-The hosted login page uses the Lock widget and is configured in the [Hosted Pages section](${manage_url}/#/login_page) in your Auth0 dashboard. To customize the look and feel of the Lock widget, see the [configuration options documentation](/libraries/lock/v10/customization).
-:::
+The Auth0 hosted login page uses the Lock widget. You can configure the page in the [Hosted Pages](${manage_url}/#/login_page) section of your Auth0 dashboard. To customize the look and feel of the Lock widget, see the [customization options documentation](/libraries/lock/v10/customization).
```
We are missing a link to the main doc https://auth0.com/docs/hosted-pages/login 

And you can omit the 2nd sentence (since the linked doc has all the info on how to configure the HLP)

mpaktiti commented on articles/quickstart/spa/_includes/_install_auth0js.md:
```diff
@@ -1,7 +1,7 @@
-Integrating Auth0 in your application requires the **auth0.js** library. Install it using npm or yarn.
-
+You need the auth0.js library to integrate Auth0 into your application.
+Install auth0.js.
```
Shouldn't this `Install auth0.js.` be more like `To install auth0.js run the following script:`

awie3di replied:
All the code in the articles has an introductory sentence in this format: "Do x.", so I felt like this doesn't need more introduction (just to keep consistency).

mpaktiti commented on articles/quickstart/spa/_includes/_login_preamble.md:
```diff
@@ -1,3 +1,8 @@
-The first step in adding authentication to your ${library} application is to provide a way for your users to log in. The fastest, most secure, and most feature-rich way to do this with Auth0 is to use the [hosted login page](https://auth0.com/docs/hosted-pages/login).
+Use the  hosted login page to provide a way for your users to log in to your ${library} application.
```
Invalid variable name, has to be escaped `${library}`

awie3di replied:
As discussed on slack, this is a placeholder that is replaced when the snippet is inserted into an article (the variable is defined in the insert command).

mpaktiti commented on articles/quickstart/spa/_includes/_login_preamble.md:
```diff
@@ -1,3 +1,8 @@
 
-When authentication is successful, three items will be returned which can all be used at some point in your application: an `access_token`, an `id_token`, and the number of seconds until the `access_token` expires.
+When the user logs in, Auth0 returns three items:
+* `access_token` - an access token
```
use this format:
        `access_token`: an access token

Also, links to the tokens docs won't be a bad idea

awie3di replied:
As discussed on slack, this is a placeholder that is replaced when the snippet is inserted into an article (the variable is defined in the insert command). 

@mpaktiti, if you could point me to the place you think I should link to, I will be happy to add the links, I'm just not sure what you mean.

mpaktiti commented on articles/quickstart/spa/_includes/_token_renewal_server_setup.md:
```diff
@@ -45,9 +51,6 @@ console.log('Listening on http://localhost:${serverPort}');
-
 ::: note
-Be sure to add `http://localhost:${serverPort}/silent` to the **Callback URLs** section in your application's client settings.
+Add `http://localhost:${serverPort}/silent` to the **Callback URLs** section in your application's client settings.
```
`to the **Callback URLs** field in your [client's settings](${manage_url}/#/clients/${account.clientId}/settings).`

mpaktiti commented on articles/quickstart/spa/_includes/_user_profile_additional_info.md:
```diff
@@ -1,3 +1,3 @@
 
-Auth0 provides a `user_metadata` object for all users. This object is intended to store information that can be modified by the user and is a useful place to keep data that is meant to be user-defined. See the [management API documentation](api/management/v2) for more information on how to allow users to make edits to their own profile information.
\ No newline at end of file
+You can allow your users to edit their profile information. Auth0 provides a `user_metadata` object to store information that users can edit. See the [management API documentation](api/management/v2) for more information.
```
This guidance looks too generic. I'd link here https://auth0.com/docs/metadata (which links to this doc https://auth0.com/docs/metadata/management-api that has some examples on how to use the Mgmt API)

mpaktiti commented on articles/quickstart/spa/_includes/_user_profile_preamble.md:
```diff
@@ -1 +1,4 @@
\ No newline at end of file
+::: panel Using the `/userinfo` endpoint
+Most applications display profile information to authenticated users. Auth0 provides a `/userinfo` endpoint for that. When you send an access token to the endpoint, it returns a JSON object with information about the user. The information stored in that object depends on how the user logged in to your application. 
+To learn more about the information returned by the `/userinfo` endpoint, see the [`/userinfo` endpoint documentation](https://auth0.com/docs/api/authentication#get-user-info).
```
make link relative

mpaktiti commented on articles/quickstart/spa/angular2/_includes/_centralized_login.md:
```diff
@@ -37,15 +46,21 @@ export class AuthService {
-Add some additional methods to the `AuthService` to fully handle authentication in the app.
+The example below shows the following methods:
+* `handleAuthentication` - looks for the result of authentication in the URL hash and processes it with the `parseHash` method from auth0.js
+* `setSession` - sets the user's access token, ID token, and the access token's expiry time 
```
Use this format for consistency:
      * `setSession`: sets the user's access token

mpaktiti commented on articles/quickstart/spa/react/_includes/_centralized_login.md:
```diff
@@ -40,14 +49,19 @@ const auth = new Auth();
 
-Add some additional methods to the `Auth` service to fully handle authentication in the app.
+The example below shows the following methods:
+* `handleAuthentication` - looks for the result of authentication in the URL hash. The method processes the hash with the `parseHash` method from auth0.js
```
Use format -> * `handleAuthentication`: looks...

mpaktiti commented on articles/quickstart/webapp/nodejs/02-user-profile.md:
```diff
@@ -16,11 +16,10 @@ budicon: 292
-Getting the user's profile information is very simple with Auth0. After the user has authenticated, a `user` object
-with the entire profile is attached to every express request. By default this will be a minimal profile, to retrieve additional profile information
-the `profile` scope should be added to the authentication scopes.
+After the users has authenticated, every express request has a `user` object with the entire user profile. 
```
`profile` does not help much (assuming that we are talking about oidc-conformant clients).

Even that will return only a few claims, see here https://auth0.com/docs/scopes/current#example-asking-for-standard-claims

If the user needs more user info they can use the Management API https://auth0.com/docs/api/management/v2#!/Users/get_users_by_id which returns sth like that:

```
{
  "email": "john.doe@gmail.com",
  "email_verified": false,
  "username": "johndoe",
  "phone_number": "+199999999999999",
  "phone_verified": false,
  "user_id": "usr_5457edea1b8f33391a000004",
  "created_at": "",
  "updated_at": "",
  "identities": [
    {
      "connection": "Initial-Connection",
      "user_id": "5457edea1b8f22891a000004",
      "provider": "auth0",
      "isSocial": false
    }
  ],
  "app_metadata": {},
  "user_metadata": {},
  "picture": "",
  "name": "",
  "nickname": "",
  "multifactor": [
    ""
  ],
  "last_ip": "",
  "last_login": "",
  "logins_count": 0,
  "blocked": false,
  "given_name": "",
  "family_name": ""
}
```

mpaktiti commented on articles/quickstart/webapp/nodejs/_includes/_login.md:
```diff
@@ -52,12 +53,24 @@ app.use(passport.session());
+
+The route creates an instance of the `auth0.WebAuth` object. Then, the route calls the `authorize` method and redirects the user to the Auth0 hosted login page.
+
+You need to make sure you get an OIDC-conformant response. You can achieve it two ways:
```
Again, I have no complete image atm, but this feels like it should be in a more prominent place for the quickstarts that use pipeline2

mpaktiti reviewed (dismissed):
Hey, I left some comments. Also, the review app is failing with this error:

```
Error compiling document articles/quickstart/spa/angular2/01-login.md: Couldn't resolve include ../../../_includes/_login_auth0_hosted_login_page included from file /app/docs/articles/quickstart/spa/angular2/_includes/_centralized_login.md
```

awie3di committed:
Rewriting the Login article in the Angular JS guide

awie3di committed:
Aligning the Nodejs Login page with the already corrected content

awie3di committed:
Changing the URL to relative

awie3di committed:
Correcting to warning

awie3di committed:
Minor correction

awie3di committed:
Changing links

awie3di committed:
Update _calling_api_create_scope.md

awie3di committed:
Correcting a link

awie3di committed:
Adding a link to the auth0 dashboard

awie3di committed:
Merge branch '3di-quickstart-updates' of https://github.com/awie3di/docs into 3di-quickstart-updates

awie3di committed:
Rewriting the User Profile article in the AngularJS quickstart

awie3di committed:
Rewriting the Calling an API article in the AngularJS quickstart

awie3di committed:
Rewriting the Authorization article in the AngularJS quickstart

awie3di committed:
FRewriting the Token Renewal article in the AngularJS quickstart

awie3di committed:
Aligning the content with the newest corrections

awie3di committed:
Correcting the AngularJS quickstart

awie3di committed:
Changing snippet to plain text

awie3di committed:
Changing snippet to plain text

awie3di committed:
Changing snippet to plain text

awie3di committed:
Correcting the note

awie3di committed:
Minor change

awie3di committed:
Correcting the link

awie3di committed:
Adding a link to documentation

awie3di committed:
Adding links

awie3di committed:
Adding links

awie3di committed:
Adding a link

awie3di committed:
Adding links to token documentation

awie3di committed:
Changing the link

awie3di committed:
Making the link relative

awie3di committed:
Minor correction

awie3di committed:
Merge branch '3di-quickstart-updates' of https://github.com/awie3di/docs into 3di-quickstart-updates

awie3di committed:
Corrections to the AngularJS Login article

awie3di committed:
Adding a link to a step

awie3di commented on articles/quickstart/spa/_includes/_auth_service_method_description_auth0js.md:
```diff
@@ -1,7 +1,30 @@
+* Audience and scope, which specify that authentication must be [OIDC-conformant](/api-auth/intro)
+* The URL where your users are redirected to after authentication.
+::: note
+In this tutorial, the route is `/callback`, which is implemented in the [Add a Callback Component](#add-a-callback-component) step. 
```
Does linking to headings work?

mpaktiti replied:
@awie3di Yes

awie3di committed:
Minor correction

awie3di committed:
Minor change

awie3di committed:
Minor change

awie3di committed:
Minor corrections

awie3di committed:
Correcting the NodeJS quickstart

awie3di committed:
Merge branch '3di-quickstart-updates' of https://github.com/awie3di/docs into 3di-quickstart-updates

awie3di committed:
Minor change

awie3di committed:
Minor change

awie3di committed:
Corrections to the AngularJS quickstart after PR

awie3di committed:
Merge branch '3di-quickstart-updates' of https://github.com/awie3di/docs into 3di-quickstart-updates

awie3di committed:
Updating note

awie3di committed:
Adding a link

awie3di committed:
Correcting the note

awie3di committed:
Correcting the Embed the Lock widget note

awie3di committed:
Removing the snippet

awie3di committed:
Minor change

awie3di committed:
Changing the order of the 'Add a callback component' and 'Process the authentication result' sections

awie3di committed:
Merge branch '3di-quickstart-updates' of https://github.com/awie3di/docs into 3di-quickstart-updates

awie3di committed:
Clarifying the sentence

awie3di committed:
Rewriting the Login article in the JavaScript quickstart

Compared to other SPA Quickstarts, this article has one section less: Add a Callback Compontent. Is that to be expected?

awie3di committed:
Rewriting the User Profile article in the JavaScript Quickstart

awie3di committed:
Rewriting the Calling an API article in the JavaScript quickstart

awie3di committed:
Minor correction

awie3di committed:
Merge branch '3di-quickstart-updates' of https://github.com/awie3di/docs into 3di-quickstart-updates

awie3di committed:
Minor corrections

awie3di committed:
Minor change

awie3di committed:
Merge branch '3di-quickstart-updates' of https://github.com/awie3di/docs into 3di-quickstart-updates

awie3di committed:
Correcting the Authorization article in theJavaScript Quickstart

awie3di committed:
Correcting the Authorization articles.

awie3di committed:
Correcting the Token Renewal article in the JavaScript Quickstart

awie3di committed:
Correcting the AngularJS Token Renewal article

awie3di committed:
Corrections

awie3di committed:
Changing the name of the Webauth object

awie3di committed:
Changing Webauth to auth0.webauth

awie3di committed:
Changing webauth to auth0.webauth

awie3di commented on articles/quickstart/spa/vanillajs/_includes/_centralized_login.md:
```diff
@@ -179,20 +190,18 @@ Provide a template with controls for the user to log in and log out.
 
 <%= include('../../_includes/_hosted_login_customization' }) %>
 
 ## Process the Authentication Result
```
@aaguiarz, In the rest of the guides, there is an 'Add a callback component' section before the 'Process the authentication result'. Shouldn't the JS quickstart contain this section as well?

awie3di committed:
Correcting a heading

awie3di committed:
Minor correction

awie3di committed:
Aligning the quickstarts

awie3di committed:
Merge branch '3di-quickstart-updates' of https://github.com/awie3di/docs into 3di-quickstart-updates

awie3di committed:
Changing a snippet to plain text

awie3di committed:
Changing a snippet to plain text

awie3di committed:
Changing a snippet to plain text

awie3di committed:
Changing a snippet to plain text

awie3di committed:
Changing snippet to plain text

awie3di committed:
Changing a snippet to plain text

awie3di committed:
Moving a note out of a snippet

awie3di committed:
Removing unnecessary snippets

awie3di committed:
Merge branch '3di-quickstart-updates' of https://github.com/awie3di/docs into 3di-quickstart-updates

aaguiarz committed:
Update _centralized_login.md

Added ../_includes to references to spa/includes (e.g. _auth_service_configure_client_details.md)

aaguiarz committed:
Update _centralized_login.md

Adding _includes/ in some include directives

awie3di committed:
Correcting minor detail

awie3di committed:
Minor correction

awie3di committed:
Minor corection

awie3di committed:
Minor correction

awie3di committed:
Making a link relative

awie3di committed:
Minor correction

awie3di committed:
minor correction

awie3di committed:
Minor correction

awie3di committed:
Removing a reduntant sentence

awie3di committed:
Changing capitalization

awie3di committed:
unifying a text link

awie3di committed:
minor correction

awie3di committed:
Adding a line break

awie3di committed:
adding a line

awie3di committed:
Changing capitalization

awie3di committed:
Minor corrections

awie3di committed:
Clarifying a sentence

awie3di committed:
Correcting capitalization

awie3di committed:
Correcting capitalization

awie3di committed:
Correcting capitalization

awie3di committed:
Minor correction

awie3di committed:
Correcting the React Token Renewal article

awie3di committed:
Updating a sentence

awie3di committed:
Updating a sentence

awie3di committed:
Updating a sentence

awie3di committed:
Updating a sentence

awie3di committed:
Merge branch '3di-quickstart-updates' of https://github.com/awie3di/docs into 3di-quickstart-updates

awie3di committed:
Aligning the section with the rest of the quickstarts

awie3di committed:
Unifying the content

awie3di committed:
Updating the note about scope names

awie3di commented on articles/quickstart/spa/_includes/_authz_preamble.md:
```diff
@@ -1,16 +1,16 @@
 
 ::: note
-Auth0 makes no assumptions about how the `scope`s for your API map to various access levels, nor is there any restriction on what you call your `scope`s and access levels. These details are at your discretion.
+Read about naming scopes and mapping them to access levels in the [Scopes documentation](/scopes). To learn more about custom scope claims, follow the [User profile claims and scope tutorial](/api-auth/tutorials/adoption/scope-custom-claims).
```
@aaguiarz, @mpaktiti, what do you think about this updated note? Are the terms correct? I used the links from Maria's comment to point the reader to detailed documentation.

awie3di committed:
Minor correction

awie3di committed:
Removing an unnecessary snippet

awie3di committed:
Removing an unnecessary snippet

awie3di committed:
Removing an unnecessary snippet

awie3di commented on articles/quickstart/spa/_includes/_user_profile_additional_info.md:
```diff
@@ -1,3 +1,3 @@
 
-Auth0 provides a `user_metadata` object for all users. This object is intended to store information that can be modified by the user and is a useful place to keep data that is meant to be user-defined. See the [management API documentation](api/management/v2) for more information on how to allow users to make edits to their own profile information.
\ No newline at end of file
+You can allow your users to edit their profile information. Auth0 provides a `user_metadata` object to store information that users can edit. See the [metadata documentation](/metadata) for more information.
```
@mpaktiti, I changed the link. Should there be more information included in this section?

awie3di committed:
Minor correction

awie3di committed:
Minor correction

awie3di committed:
Simplifying the sentence

awie3di committed:
Simplifying a sentence

awie3di committed:
Simplifying a sentence

awie3di committed:
Simplifying a sentence

awie3di committed:
Minor correction

awie3di committed:
Simplifying a sentence

awie3di committed:
Simplifying a sentence

awie3di committed:
Simplifying a sentence

awie3di committed:
Simplifying a sentence

awie3di committed:
Simplifying a sentence

awie3di committed:
Simplifying a sentence

awie3di committed:
Simplifying a sentence

awie3di committed:
Minor correction

awie3di committed:
Minor correction

awie3di committed:
Minor correction

awie3di committed:
Minor correction

awie3di committed:
Simplifying a sentence

awie3di committed:
Minor correction

awie3di committed:
Simplifying a sentence

awie3di committed:
Simplifying a sentence

awie3di committed:
Simplifying a sentence

aaguiarz committed:
Update _authz_determining_scopes.md

Reordering the context to avoid redundancy and keep meaning

aaguiarz committed:
Update 04-authorization.md

aaguiarz committed:
Update _authz_determining_scopes.md

aaguiarz committed:
Update _authz_determining_scopes.md

aaguiarz committed:
Update 04-authorization.md

aaguiarz committed:
Update _authz_user_has_scopes.md

aaguiarz committed:
Update 04-authorization.md

aaguiarz committed:
Update _authz_client_routes_disclaimer.md

aaguiarz committed:
Update _login_preamble.md

Added the whole content of 'Add Authentication with Auth0' section

aaguiarz committed:
Used new '_login_preamble.md'

aaguiarz committed:
Adjusted it to use the new _login_preamble.md

aaguiarz committed:
Adjusted content to use new _login_preamble

aaguiarz committed:
Adjusted to use the new _login_preamble

aaguiarz committed:
Conditional for Embedded Login

aaguiarz committed:
Added links to related documents

aaguiarz committed:
Update _login_preamble.md

aaguiarz committed:
Update _login_preamble.md

aaguiarz committed:
Update _getting_started.md

aaguiarz committed:
Update _getting_started.md

aaguiarz committed:
Update _getting_started.md

aaguiarz committed:
Update _authz_preamble.md

aaguiarz committed:
Update _calling_api_preamble.md

aaguiarz committed:
Update _user_profile_in_memory.md

aaguiarz committed:
Update _centralized_login.md

aaguiarz committed:
Removed redundant content

aaguiarz committed:
Update _callback_component.md

Removed the panel

aaguiarz committed:
Update _centralized_login.md

aaguiarz committed:
Update _centralized_login.md

aaguiarz committed:
Update 04-authorization.md

aaguiarz committed:
Update _token_renewal_server_setup.md

aaguiarz committed:
Update _centralized_login.md

aaguiarz committed:
Update _centralized_login.md

Removed 'About the Authentication Service'

aaguiarz committed:
Update _user_profile_in_memory.md

Removed the service name to be able to use it for angularjs too

aaguiarz committed:
Update 03-calling-an-api.md

Fixed error in the 'set the audience and scope'

aaguiarz committed:
Update _authz_determining_scopes.md

Removing sentence that talks about auth0.webAuth so it can be applied to AngularJS

aaguiarz committed:
Update _authz_set_session.md

aaguiarz committed:
Update _authz_set_session.md

aaguiarz committed:
Update _centralized_login.md

Added 'checkpoint' back.

aaguiarz committed:
Update _centralized_login.md

aaguiarz committed:
Update 04-authorization.md

aaguiarz committed:
Update _calling_api_create_api.md

removed reference to auth0.WebAuth because some SPA samp;es do not use it (e.g. AngularJS)

aaguiarz committed:
Update _getting_started.md

aaguiarz committed:
Update _centralized_login.md

aaguiarz committed:
Update _centralized_login.md

aaguiarz committed:
Update _user_profile_in_memory.md

aaguiarz committed:
Update 02-user-profile.md

aaguiarz committed:
Update 02-user-profile.md

aaguiarz committed:
Update 02-user-profile.md

aaguiarz committed:
Update 02-user-profile.md

aaguiarz committed:
Update 04-authorization.md

aaguiarz committed:
Update 04-authorization.md

aaguiarz committed:
Update 04-authorization.md

aaguiarz committed:
Update 04-authorization.md

aaguiarz committed:
Update 04-authorization.md

aaguiarz committed:
Update _getting_started.md

aaguiarz committed:
Update _centralized_login.md

aaguiarz committed:
Update _login_preamble.md

aaguiarz committed:
Update _centralized_login.md

aaguiarz committed:
Update _centralized_login.md

aaguiarz committed:
Update _authz_set_session.md

aaguiarz committed:
Update 02-user-profile.md

'stringify' refers to the Javascript function, it's easier to understand what it means than saying 'put the object in a string'.

aaguiarz committed:
Merge branch 'master' into 3di-quickstart-updates

aaguiarz committed:
Update _getting_started.md

aaguiarz committed:
Update _getting_started.md
